seps/sep-0002.md

```diff
@@ -0,0 +1,169 @@
+
+A primary drawback to this graph-based approach is that there is no canonical source of group membership. A single group of N members can have M distinct versions of who belongs in that group, where M is the number of group members who decline to "mutually follow" each newly requested group member.  Group membership is managed by an object on the namespace of anyone who has requested access to a group, which contains a group identifier as a key with a value of a list of accountIds (AccountList). Even if no one approves someone's request to join a group, the requestor will retain a valid instance of that groupIdentifer:AccountList on their socialDB namespace - with their accountID included as a group member. This results in muliple versions of graphs which represent the definition of members of a group. For more details on this implementation, see minute 33:45 [here](https://youtu.be/HAyweTF7IjU?si=z_hfct3oK_86nt_o)
+
+With this approach, can we efficiently determine which group members should receive a notification when a post is created on that group's page?  How challenging would it be to ensure that only group members can post to this group's page? There is a solution here, however these implicitly defined group connections with multiple versions of group membership adds complexity. 
+
+
+1. [Communities on DevHub](https://near.org/devgovgigs.near/widget/Ideas)
+
+DevHub's community implementation exists on a distinct contract from near.social. This results in a lack of discoverability of DevHub's groups across gateways due to its community data not being included in the indexes which power search on socialdB.
+
+
+## Specification 
+Building upon the NFT-based groups implementation, we would benefit from the following the creation and maintanence of two [Core Social Indexers](https://docs.google.com/document/d/14OmYzMYX60lBm2U2Ridz6GZCeN_A6IcU_qn-yPwLTfs/edit#heading=h.nfrlp9tpavvv):
+
+1. A dataplatform.near Core Social indexer for NFT_Groups, that tracks posts to main which include a groupContractId, e.g. writes to post.main.[groupContractId], and provides an API which lists all of the groups which have been created in socialDB. 
+	1. To support the ability to search for groups, where search text is matched against the group's metadata such as name, symbol, description, and spec, the above indexer should fetch and store this metadata from a source such as the [Mintbase API](https://cloud.hasura.io/public/graphiql?endpoint=https%3A%2F%2Fgraph.mintbase.xyz%2Fmainnet&header=mb-api-key%3Aanon&query=query+MyQuery+%7B%0A++nft_metadata%28%0A++++where%3A+%7Breference%3A+%7B_eq%3A+%22nb0-oBR379DzoFYeYv-LesjVNmrVlDs5IqQ8hfDfnMU%22%7D%7D%0A++%29+%7B%0A++++id%0A++++media%0A++++reference_blob%0A++%7D%0A%7D%0A
+
+1. A dataplatform.near Core Social indexer for Group_Members, that maintains a list of NFT owners for each NFT_Group, and provides an API which serves a list of all AccountIds which belong to a group.
```

Comment:
I'm wondering how a group membership should be tracked. We will index posts so one natural way is to see if a post in the group was created. Is there any other way that this could be achieved?